### PR TITLE
Fix forwarding on token create on perf standby when an alias is created

### PR DESCRIPTION
**Pull request by banks**
OSS part. The whole change including tests has been approved in [Enterprise](https://github.com/hashicorp/vault-enterprise/pull/4148).

### CODE REVIEW

File: website/content/docs/upgrading/upgrade-to-1.13.x.mdx [hunk 1]
```diff
@@ -90,6 +92,20 @@ forwarded to the active node of this cluster.
 
 As a workaround, submit revocation requests to the active node only.
 
+### Slow Startup Time When Storing PKI Certificates
```
Comment (ncabatoff) [inline website/content/docs/upgrading/upgrade-to-1.13.x.mdx:95]: Why is this part of this PR?

File: vault/token_store.go [hunk 2]
```diff
@@ -2872,10 +2872,20 @@ func (ts *TokenStore) handleCreateCommon(ctx context.Context, req *logical.Reque
 			MountType:     mountValidationResp.Type,
 		}
 
-		// Create or fetch entity from entity alias
+		// Create or fetch entity from entity alias. Note that we might be on a perf
+		// secondary so a create would return a ReadOnly error which would cause an
```
Comment (mpalmi) [inline vault/token_store.go:2876]: Should this be "perf standby"?

File: website/content/docs/upgrading/upgrade-to-1.13.x.mdx (hunk 1 above)
Comment (banks) [reply website/content/docs/upgrading/upgrade-to-1.13.x.mdx:95]: 🤔 Oh it was a merge conflict from Ent but I seem to have put this back in the wrong place. Let me fix that.

Comment (ncabatoff) [reply website/content/docs/upgrading/upgrade-to-1.13.x.mdx:95]: nvm, missed that it was just moved.

File: vault/token_store.go [hunk 3]
```diff
@@ -2872,10 +2872,20 @@ func (ts *TokenStore) handleCreateCommon(ctx context.Context, req *logical.Reque
...
+		// RPC-based redirect. That path doesn't register leases since the code that
+		// calls RegisterAuth is in the http layer... So be careful to catch and
+		// handle readonly ourselves.
 		entity, _, err := ts.core.identityStore.CreateOrFetchEntity(ctx, alias)
 		if err != nil {
-			return nil, err
+			auth := &logical.Auth{
+				Alias: alias,
+			}
+			entity, _, err = possiblyForwardAliasCreation(ctx, ts.core, err, auth, entity)
```
Comment (mpalmi) [inline vault/token_store.go:2885]: Neat. I had a note about handling the active/perf primary case, but it looks like this wrapper handles that.

**Review by ncabatoff — APPROVED** [review]

**Review by mpalmi — COMMENTED** [review]
Just one nit. Seems good to me, but maybe we should wait for other reviews just in case.

**Review by hghaf099 — APPROVED** [review]
Looks good to me!

**Pull request merged**